coil-base/src/test/java/coil/fetch/HttpFetcherTest.kt

```diff
@@ -113,4 +119,17 @@ class HttpFetcherTest {
         val body5 = ResponseBody.create(null, byteArrayOf())
         assertNull(fetcher.getMimeType(url5, body5))
     }
+
+    @Test
+    fun `not found response is cached`() {
+        val uri = createMockWebServer(context).url("/normal.jpg").toString().toUri()
+        val fetcher = HttpUriFetcher(callFactory)
+
+        assertFailsWith<HttpException> {
+            runBlocking {
+                fetcher.fetch(pool, uri, PixelSize(100, 100), createOptions(context))
+            }
+        }
+        assertTrue(cache.urls().hasNext())
```

Comment:
Can you change this to assert that the `url` we fetched is the same as the one in the cache?

Reply:
Done